### ISSUE TITLE
AssertError:buf should have data from resbuf, shutting down process

### ISSUE THREAD

**Issue by hacql2004** [Raymw/Federated-XGBoost#1]
Hi, thanks for your sharing.
When I try to replica your demo, I met a error like "AssertError:buf should have data from resbuf, shutting down process", "raise RuntimeError('Get nonzero return code=%d on %s %s' % (ret, cmd, env))". 
This error is occured after type the below command:
"python ../../xgboost/dmlc-core/tracker/dmlc-submit --cluster local --num-workers 2 python main.py"
Here is the error detail.
![image](https://user-images.githubusercontent.com/21165759/75020872-7b7ce400-54ce-11ea-8974-cbc68b2c7f6b.png)

Besides, when I run your demo in single worker model, it works well.

Could you help me to figure out it ? Thanks in advance.

**Comment by Raymw** [comment]
hi @hacql2004, I run the code again by using the command shown above. It works well on my computer and also I didn't meet this error before. Did you set the rabit ?  BTW, I am using python3.6 and I am not sure whether there is incompatibility between 3.6 and 3.7. Thanks :)

**Comment by hacql2004** [comment]
@Raymw thanks for your active reply. I've changed my envs into python3.6, while the error continues.  My xgboost was modified and rebuilt  from its newest master branch. I tried to just replace your 'para.h', 'updater_histmaker.cc' directily, but make failed. Thus, I modified the related part of files manually with your instruction. Could u share your xgboost branch edition. or any other recommendation? Thanks.

![image](https://user-images.githubusercontent.com/21165759/75134408-313b7300-5719-11ea-8010-7729c75a05d5.png)
